core/src/types/storage.rs

```diff
@@ -1161,48 +1159,25 @@ impl Default for Epochs {
     /// block height 0.
     fn default() -> Self {
         Self {
-            first_known_epoch: Epoch::default(),
             first_block_heights: vec![BlockHeight::default()],
         }
     }
 }
 
 impl Epochs {
-    /// Record start of a new epoch at the given block height and trim any
-    /// epochs that ended more than `max_age_num_blocks` ago.
-    pub fn new_epoch(
-        &mut self,
-        block_height: BlockHeight,
-        max_age_num_blocks: u64,
-    ) {
-        let min_block_height_to_keep = (block_height.0 + 1)
-            .checked_sub(max_age_num_blocks)
-            .unwrap_or_default();
-        // trim off any epochs whose last block is before the limit
-        while let Some((_first_known_epoch_height, rest)) =
-            self.first_block_heights.split_first()
-        {
-            if let Some(second_known_epoch_height) = rest.first() {
-                if second_known_epoch_height.0 < min_block_height_to_keep {
-                    self.first_known_epoch = self.first_known_epoch.next();
-                    self.first_block_heights = rest.to_vec();
-                    continue;
-                }
-            }
-            break;
-        }
+    /// Record start of a new epoch at the given block height
+    pub fn new_epoch(&mut self, block_height: BlockHeight) {
         self.first_block_heights.push(block_height);
     }
 
-    /// Look-up the epoch of a given block height.
+    /// Look up the epoch of a given block height.
+    /// TODO: handle passing of a future block height? (only valid for current
+    /// or past)
```

Comment:
We can't determine future epochs because epoch progression depends on 2 params - min. num of blocks and min duration. I think if the given `block_height > current_height` we can return `None`

Reply:
ah, nvm, we don't have enough context here to check current height. The current behavior is that we return the current epoch, which is probably fine

Reply:
Yeah, I put this here because it doesn't rly make sense to pass a future block height into this, but figured if this function were to eventually also take in the current block height, we could return `None` perhaps